framework/wazuh/core/agent.py

```diff
@@ -1028,13 +1030,13 @@ def add_group_to_agent(group_id, agent_id, force=False, replace=False, replace_l
                 raise WazuhError(1752)
         else:
             # Check if the group already belongs to the agent
-            if group_id in multigroup_name.split(','):
+            if group_id in agent_groups:
                 raise WazuhError(1751)
 
-            multigroup_name = (multigroup_name + ',' if multigroup_name else '') + group_id
+            multigroup_name = f'{multigroup_name}{"," if multigroup_name else ""}{group_id}'
 
         # Check multigroup limit
-        if Agent.check_multigroup_limit(agent_id):
+        if len(agent_groups) > common.max_groups_per_multigroup:
```

Comment:
This behavior is different from the previous one. Before looking only at the code, there could be repeated group names, and these were counted to see the maximum limit. Now it is a set, so in case of repeats, they are eliminated and do not count for the maximum limit.